packages/frameworks-sveltekit/src/lib/index.ts

```diff
@@ -350,3 +465,26 @@ export function setEnvDefaults(envObject: any, config: SvelteKitAuthConfig) {
     return finalProvider
   })
 }
+
+/**
+ * Extract the origin and base path from either `AUTH_URL` or `NEXTAUTH_URL` environment variables,
+ * or the request's headers and the {@link NextAuthConfig.basePath} option.
+ */
+export function createActionURL(
+  action: AuthAction,
+  h: Headers,
+  basePath?: string
+) {
+  const envUrl = process.env.AUTH_URL ?? process.env.NEXTAUTH_URL
+  if (envUrl) {
+    const { origin, pathname } = new URL(envUrl)
+    const separator = pathname.endsWith("/") ? "" : "/"
+    return new URL(`${origin}${pathname}${separator}${action}`)
+  }
+  const host = h.get("x-forwarded-host") ?? h.get("host")
+  const protocol = h.get("x-forwarded-proto") === "http" ? "http" : "https"
+  // @ts-expect-error `basePath` value is default'ed to "/api/auth" in `setEnvDefaults`
+  const { origin, pathname } = new URL(basePath, `${protocol}://${host}`)
+  const separator = pathname.endsWith("/") ? "" : "/"
+  return new URL(`${origin}${pathname}${separator}${action}`)
+}
```

Comment:
Could simplify a bit. See: https://github.com/nextauthjs/next-auth/blob/30fb90f9159d8375826695351cf0a4cd490885f4/packages/next-auth/src/lib/actions.ts#L113-L129

`process.env` is not available in SvelteKit, so that needs adjusting.